server/src/metadata/field-metadata/services/field-metadata.service.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+import { TypeOrmQueryService } from '@ptc-org/nestjs-query-typeorm';
+
+import { FieldMetadata } from 'src/metadata/field-metadata/field-metadata.entity';
+import {
+  convertFieldMetadataToColumnChanges,
+  convertMetadataTypeToColumnType,
+  generateColumnName,
+  generateTargetColumnMap,
+} from 'src/metadata/field-metadata/utils/field-metadata.util';
+import { MigrationRunnerService } from 'src/metadata/migration-runner/migration-runner.service';
+import { TenantMigrationService } from 'src/metadata/tenant-migration/tenant-migration.service';
+import { ObjectMetadataService } from 'src/metadata/object-metadata/services/object-metadata.service';
+import {
+  TenantMigrationColumnChange,
+  TenantMigrationTableChange,
+} from 'src/metadata/tenant-migration/tenant-migration.entity';
+
+@Injectable()
+export class FieldMetadataService extends TypeOrmQueryService<FieldMetadata> {
```

Comment:
We have to extend TypeOrmQueryService here, this will allow us to override methods that are called by auto-generated resolvers such as createOne.
This is useful here because we want to add extra treatment such as validating the request (do we have this object in the DB? Do we have access to it? Do we have a duplicate field already) however I don't think this is the right place to do such things.
It also allows us to run the migrations inside the tenant once the field is created. Again, this might not be the right place and that something we could do later with a scheduler/listener/job system

Reply:
@Weiko Actually it's the right place to check logical stuff in services. The only part that I think can be moved, is the migration part.
We can definitely use a subscription system, or nest.js event system https://docs.nestjs.com/techniques/events for the migration part.